src/Features/Core/Portable/Copilot/ICopilotChangeAnalysisService.cs

```diff
@@ -59,27 +66,290 @@ public async Task<CopilotChangeAnalysis> AnalyzeChangeAsync(
+        {
+            // Get diagnostics in parallel for all edited spans, for the desired diagnostic kind.
+            return ProducerConsumer<DiagnosticData>.RunParallelAsync(
+                newSpans,
+                static async (span, callback, args, cancellationToken) =>
+                {
+                    var (newDocument, diagnosticKind) = args;
+                    var diagnosticAnalyzerService = newDocument.Project.Solution.Services.GetRequiredService<IDiagnosticAnalyzerService>();
+                    var diagnostics = await diagnosticAnalyzerService.GetDiagnosticsForSpanAsync(
+                        newDocument, span, diagnosticKind, cancellationToken).ConfigureAwait(false);
+                    foreach (var diagnostic in diagnostics)
+                    {
+                        // Ignore supressed and hidden diagnostics.  These are things the user has said they do not
+                        // care about and would then have no interest in being auto fixed.
+                        if (IsVisibleDiagnostic(diagnostic.IsSuppressed, diagnostic.Severity))
+                            callback(diagnostic);
+                    }
+                },
+                args: (newDocument, diagnosticKind),
+                cancellationToken);
+        }
+    }
+
+    private static bool IsVisibleDiagnostic(bool isSuppressed, DiagnosticSeverity severity)
+        => !isSuppressed && severity != DiagnosticSeverity.Hidden;
+
+    private async Task<CopilotCodeFixAnalysis> ComputeCodeFixAnalysisAsync(
+        Document newDocument,
+        ArrayBuilder<TextSpan> newSpans,
+        CancellationToken cancellationToken)
+    {
+        // Determine how long it would be to even compute code fixes for these changed regions.
+        var totalComputationStopWatch = SharedStopwatch.StartNew();
+        var codeFixCollections = await ComputeCodeFixCollectionsAsync().ConfigureAwait(false);
+        var totalComputationTime = totalComputationStopWatch.Elapsed;
+
+        var diagnosticIdToCount = new Dictionary<string, int>();
+        var diagnosticIdToApplicationTime = new Dictionary<string, TimeSpan>();
+        var diagnosticIdToProviderName = new Dictionary<string, HashSet<string>>();
+        var providerNameToApplicationTime = new Dictionary<string, TimeSpan>();
+
+        var totalApplicationTimeStopWatch = SharedStopwatch.StartNew();
+        await ProducerConsumer<(CodeFixCollection collection, TimeSpan elapsedTime)>.RunParallelAsync(
+            codeFixCollections,
+            produceItems: static async (codeFixCollection, callback, args, cancellationToken) =>
+            {
+                var (@this, solution, _, _, _, _) = args;
+                var firstAction = GetFirstAction(codeFixCollection.Fixes[0]);
+
+                var applicationTimeStopWatch = SharedStopwatch.StartNew();
+                var result = await firstAction.GetPreviewOperationsAsync(solution, cancellationToken).ConfigureAwait(false);
```

Comment:
Are we calling `GetPreviewOperationsAsync` just to ensure that the method would return? It does not appear that we are using the result.

Reply:
we are computing how long it would take to actually perform the fix.